components/gateway/cmd/main.go

```diff
@@ -105,16 +105,14 @@ func initAuditLogsSvc(done chan bool) AuditogService {
 	err := envconfig.InitWithPrefix(&auditlogCfg, "APP")
 	exitOnError(err, "Error while loading auditlog cfg")
 
-	httpClient, tenant := initAuditlogClient(auditlogCfg)
-	uuidSvc := uuid.NewService()
-	timeSvc := &time.TimeService{}
+	httpClient, msgFactory := initAuditlogClient(auditlogCfg)
 
-	auditlogClient, err := auditlog.NewClient(auditlogCfg, httpClient, uuidSvc, timeSvc, tenant)
+	auditlogClient, err := auditlog.NewClient(auditlogCfg, httpClient)
 	exitOnError(err, "Error while creating auditlog client from cfg")
 
-	auditlogMsgChannel := make(chan auditlog.AuditlogMessage)
+	auditlogMsgChannel := make(chan auditlog.Message)
```

Comment:
make this channel buffered, thanks to that your producer won't be blocked by the consumer. 
In addition to that, modify this code to avoid deadlocks:
```
func (sink *Sink) Log(request, response string, claims proxy.Claims) error {
	msg := Message{
		Request:  request,
		Response: response,
		Claims:   claims,
	}
	sink.logsChannel <- msg
	return nil
}
```
There can be a situation that no one reads from the channel, and even it will be buffered channel, at some moment this operation will block everything. Sending to the channel should be done with some timeout (pseudocode):
```
select {
  case sink.logsChannel <- msg:
 case <- time.After(time.Second*5):
   return errors.New("oh no, something is really wrong")
}
```